Nos/Models/NosNotification+CoreDataClass.swift

```diff
@@ -76,4 +70,15 @@ public class NosNotification: NSManagedObject {
             try? context.saveIfNeeded()
         }
     }
+
+    static func oldNotificationsRequest() -> NSFetchRequest<NosNotification> {
```

Comment:
I think `static` is probably the way to go here; should we change all the `class func`s in here to be `static func`s for consistency?

Reply:
I usually use `class` because unlike `static` class functions can be overridden in a subclass, which comes in handy when mocking for unit tests. Why do you prefer `static`?

Reply:
Lately I've been trying to keep things locked down a bit more, so in this case that would mean not allowing it to be overridden. Recently I've gotten away from subclassing in order to mock; instead, I've used protocols and created a mock implementation. I'm not sure if that's doable here since I know this is a NSManagedObject subclass, but maybe? And even if it is doable, I understand that it might not be something we want to do right now. I'm happy with `class` as long are we're consistent.

Reply:
That makes sense. No harm in going with `static` I guess and changing it to `class` if we need to override it. In general these types of access controls within an app (as opposed to an SDK) seem unnecessary to me, but if you find them helpful then let's use them.